edgelet/edgelet-docker/src/runtime.rs

```diff
@@ -590,6 +597,95 @@ impl ModuleRuntime for DockerModuleRuntime {
         )
     }
 
+    fn system_resources(&self) -> Self::SystemResourcesFuture {
+        info!("Querying system resources...");
+        let mut system_info = sysinfo::System::new();
+        system_info.refresh_all();
+
+        let client = self.client.clone();
+        let docker_stats = self
+            .list() // Get all modules
+            .and_then(|modules: Vec<Self::Module>| { // Get iterable of stats
+                future::join_all(
+                    modules
+                        .into_iter()
+                        .map(|module| module.name().to_owned())
+                        .map(move |name| {
+                            client
+                                .container_api()
+                                .container_stats(&name, false)
```

Comment:
This will fail if the container happens to be deleted after the list call that returned it but before it can be `container_stats`ed. We shouldn't fail the whole `system_resources` API in that case, just skip that container.

Regular `list` had the same problem, which is why we added `list_with_details` that handles the container-not-found error. You should do the same thing here.